app/src/main/java/com/moviefiend/torian/moviefiend/MainActivity.java

```diff
@@ -1,13 +1,26 @@
 package com.moviefiend.torian.moviefiend;
 
+import android.content.Intent;
 import android.support.v7.app.AppCompatActivity;
 import android.os.Bundle;
 
-public class MainActivity extends AppCompatActivity {
+import com.moviefiend.torian.moviefiend.network.NowPlayingResponse;
+
+public class MainActivity extends AppCompatActivity implements MovieListFragment.Listener {
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+
+        MovieListFragment fragment = (MovieListFragment) getFragmentManager().findFragmentById(R.id.movie_list_fragment);
+        fragment.setListener(this);
```

Comment:
@ifeins you are relating to a new scenario. Before you talked about the listener becoming invalid and now you talk about it not being set. In that case there is no problem, because the method will not be called (I'm checking for null especially for that) so you will have a bug, but that is because you used the API wrong